graylog2-web-interface/src/components/indices/IndexSetTemplates/SelectIndexSetTemplateModal.tsx

```diff
@@ -0,0 +1,216 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import { PluginStore } from 'graylog-web-plugin/plugin';
+
+import useSendTelemetry from 'logic/telemetry/useSendTelemetry';
+import useLocation from 'routing/useLocation';
+import { getPathnameWithoutId } from 'util/URLUtils';
+import { TELEMETRY_EVENT_TYPE } from 'logic/telemetry/Constants';
+import { Row, Col, Modal, SegmentedControl } from 'components/bootstrap';
+import { ModalSubmit, Spinner, Select } from 'components/common';
+import useSelectedIndexSetTemplate from 'components/indices/IndexSetTemplates/hooks/useSelectedTemplate';
+import useTemplates from 'components/indices/IndexSetTemplates/hooks/useTemplates';
+import TemplateDetails from 'components/indices/IndexSetTemplates/TemplateDetails';
+import IndexSetTemplateCard from 'components/indices/IndexSetTemplates/IndexSetTemplateCard';
+import type { IndexSetTemplate } from 'components/indices/IndexSetTemplates/types';
+import { DATA_TIERING_TYPE } from 'components/indices/data-tiering';
+import type { Sort } from 'stores/PaginationTypes';
+
+type Props = {
+  show: boolean,
+  hideModal: () => void,
+}
+
+type TemplateCategorySegment = 'built_in' | 'custom';
+
+const FlexWrapper = styled.div(({ theme }) => css`
+  display: flex;
+  flex-direction: column;
+  gap: ${theme.spacings.md};
+`);
+
+const SelectIndexSetTemplateModal = ({ hideModal, show }: Props) => {
+  const { selectedIndexSetTemplate, setSelectedIndexSetTemplate } = useSelectedIndexSetTemplate();
+  const [tempSelectedTemplate, setTempSelectedTemplate] = useState<IndexSetTemplate | undefined>(selectedIndexSetTemplate);
+  const initialTemplateCategory = (selectedIndexSetTemplate && !selectedIndexSetTemplate.built_in) ? 'custom' : 'built_in';
+  const [selectedTemplateCategory, setSelectedTemplateCategory] = useState<TemplateCategorySegment>(initialTemplateCategory);
+  const sendTelemetry = useSendTelemetry();
+  const { pathname } = useLocation();
+  const telemetryPathName = useMemo(() => getPathnameWithoutId(pathname), [pathname]);
+
+  const templateCategorySegments: Array<{value: TemplateCategorySegment, label: string}> = [
+    { value: 'built_in', label: 'Default Templates' },
+    { value: 'custom', label: 'Custom Templates' },
+  ];
+
+  const sort: Sort = {
+    attributeId: 'title',
+    direction: 'asc',
+  };
+
+  const {
+    isLoading: isLoadingBuiltIn,
+    data: { list: builtInList },
+  } = useTemplates(
+    {
+      page: 1,
+      pageSize: 3,
+      query: 'built_in:true',
+      sort: sort,
+    }, { enabled: true },
```

Comment:
We can add the default value `enabled = true`  to `useTemplates`, so we add the param only when we want to disable it.